File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/gameLogs/types/HealDoneModel.ts
export type HealDoneModel = {
    id: number;
    spell: string;
    value: number;
    overheal: number;
    time: string;
    creator: string;
    target: string;
    isCrit: boolean;
    isAbsorbed: boolean;
    combatPlayerId: number;
}

File: src/API/CombatAnalysis.CombatParserAPI/Models/HealDoneModel.cs
using System.ComponentModel.DataAnnotations;

namespace CombatAnalysis.CombatParserAPI.Models;

public class HealDoneModel
{
    [Range(0, int.MaxValue)]
    public int Id { get; set; }

    [Required]
    public string Spell { get; set; }

    [Range(0, int.MaxValue)]
    public int Value { get; set; }

    [Range(0, int.MaxValue)]
    public int Overheal { get; set; }

    [Required]
    public TimeSpan Time { get; set; }

    [Required]
    public string Creator { get; set; }

    [Required]
    public string Target { get; set; }

    [Required]
    public bool IsCrit { get; set; }

    [Required]
    public bool IsAbsorbed { get; set; }

    [Range(0, int.MaxValue)]
    public int CombatPlayerId { get; set; }
}
